### PR TITLE
Remove getPageLayoutBox, getLayoutBox, and getIntersectionChangeEntry APIs from docking

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #31540.

A more holistic rewrite could eliminate significant amount of measures. However, in my analysis:

1. Most of the measurements (and most performance sensitive ones) happen as the result of `PositionObserver` callback. However, that callback, conveniently, provides the `ClientRect` structure, which I used when possible to eliminate most of the measurements.
2. The code as it stands is fairly efficient. E.g. it starts monitoring of positions only when the element is in a right corridor.

### CODE REVIEW

File: extensions/amp-video-docking/0.1/amp-video-docking.js [hunk 1]
```diff
@@ -594,8 +596,9 @@ export class VideoDocking {
     }
 
     const {element} = video;
-    const {intersectionRect} = element.getIntersectionChangeEntry();
-    if (!isSizedRect(intersectionRect)) {
+    const clientRect = element.getBoundingClientRect();
+    const intersectionRect = rectIntersection(clientRect, this.viewportRect_);
+    if (!intersectionRect || !isSizedRect(intersectionRect)) {
```
Comment (samouri) [inline extensions/amp-video-docking/0.1/amp-video-docking.js:601]: We don't have a better way to say "is element within viewport"?

File: extensions/amp-video-docking/0.1/amp-video-docking.js [hunk 2]
```diff
@@ -725,43 +730,46 @@ export class VideoDocking {
...
    * @return {!Promise}
    * @private
    */
-  dockInTransferLayerStep_(video, target, opt_step) {
+  dockInTransferLayerStep_(video, target, step, opt_clientRect) {
     // Do this in a multi-step process due to a browser quirk in transferring
     // layers to GPU.
-    const step = opt_step || 0.1;
     // This cutoff is arbitrary and may be dependant on performance.
-    // TODO(alanorozco): Investigate.
```
Comment (samouri) [inline extensions/amp-video-docking/0.1/amp-video-docking.js:753]: Do you know what all this "step" logic is about?

**Review by samouri — APPROVED** [review]
Assuming you've tested the functionality, code LGTM

File: extensions/amp-video-docking/0.1/amp-video-docking.js (hunk 1 above)
Comment (dvoytenko) [reply extensions/amp-video-docking/0.1/amp-video-docking.js:601]: No better "sync" way. I tried to keep the structure as much as possible intact. There's a v2 version of this API in works and we'll need to make sure we just use InOb for most of the needs there.

File: extensions/amp-video-docking/0.1/amp-video-docking.js (hunk 2 above)
Comment (dvoytenko) [reply extensions/amp-video-docking/0.1/amp-video-docking.js:753]: I don't understand it 100%, but it appears it tries to sync animation with scrolling in a limited way. That way the animation is more seamless with scrolling.

**Pull request merged**